ada/ast.py

```diff
@@ -14800,6 +14827,16 @@ def expr_type():
 
     env_spec = EnvSpec(add_to_env_kv(Self.name_symbol, Self))
 
+    @langkit_property()
+    def is_constant_object():
+        loop_spec = Var(Entity.parent.cast(ForLoopSpec))
+        return If(
+            loop_spec.loop_type.is_a(IterType.alt_of),
+            loop_spec.iter_expr.cast_or_raise(Name).is_constant_reference,
+            # IterType.alt_in are always constant
```

Comment:
You should probably put a larger comment explaining the different cases that we went over live the other day